RUFAS/biophysical/animal/reproduction/reproduction.py

```diff
@@ -1,2 +1,2768 @@
+    conceptus_weight : float, optional
+        Weight of the conceptus, by default 0.0.
+    calf_birth_weight : float, optional
+        Birth weight of the calf, by default 0.0.
+    calves : int, optional
+        Number of calves, by default 0.
+    calving_interval : int, optional
+        Interval between calvings, by default 0.
+    calving_interval_history : list[int], optional
+        History of calving intervals, by default None.
+    body_weight_at_calving : float, optional
+        Body weight of the animal at calving, by default 0.0.
+    do_not_breed : bool, optional
+        Flag indicating if the animal should not breed, by default False.
+    """
+
+    def __init__(
+            self,
+            heifer_reproduction_program: HeiferReproductionProtocol = None,
+            heifer_reproduction_sub_program: HEIFER_REPRODUCTION_SUB_PROTOCOLS = None,
+            cow_reproduction_program: CowReproductionProtocol = None,
+            cow_presynch_program: CowPreSynchSubProtocol = None,
+            cow_ovsynch_program: CowTAISubProtocol = None,
+            cow_resynch_program: CowReSynchSubProtocol = None,
+            ai_day: int = 0,
+            estrus_day: int = 0,
+            abortion_day: int = 0,
+            breeding_to_preg_time: int = 0,
+            conception_rate: float = 0.0,
+            cow_TAI_conception_rate: float = 0.0,
+            num_conception_rate_decreases: int = 0,
+            hormone_schedule: dict[int, dict[str, Any]] = None,
+            gestation_length: int = 0,
+            conceptus_weight: float = 0.0,
+            calf_birth_weight: float = 0.0,
+            calves: int = 0,
+            calving_interval: int = 0,
+            calving_interval_history: list[int] = None,
+            body_weight_at_calving: float = 0.0,
+            do_not_breed: bool = None,
+            estrus_count: int = 0
+    ) -> None:
+        self.heifer_reproduction_program = HeiferReproductionProtocol(AnimalConfig.heifer_reproduction_program) \
+            if heifer_reproduction_program is None else heifer_reproduction_program
+        self.heifer_reproduction_sub_program = AnimalConfig.heifer_reproduction_sub_program \
+            if heifer_reproduction_sub_program is None else heifer_reproduction_sub_program
+        self.cow_reproduction_program = CowReproductionProtocol(AnimalConfig.cow_reproduction_program) \
+            if cow_reproduction_program is None else cow_reproduction_program
+        self.cow_presynch_program = CowPreSynchSubProtocol(AnimalConfig.cow_presynch_method) \
+            if cow_presynch_program is None else cow_presynch_program
+        self.cow_ovsynch_program = CowTAISubProtocol(AnimalConfig.cow_tai_method) \
+            if cow_ovsynch_program is None else cow_ovsynch_program
+        self.cow_resynch_program = CowReSynchSubProtocol(AnimalConfig.cow_resynch_method) \
+            if cow_resynch_program is None else cow_resynch_program
+
+        self.ai_day = ai_day if ai_day else 0
```

Comment:
```suggestion
        self.ai_day = ai_day
```
I think since this is defaulted to 0 in the params, you don't need to have the else statement.

Same would apply to all the other attributes that are defaulted to 0 in the params.